exporters/otlpjson/build.gradle

```diff
@@ -0,0 +1,22 @@
+plugins {
+    id "java-library"
+    id "maven-publish"
+
+    id "ru.vyarus.animalsniffer"
+}
+
+description = 'OpenTelemetry Protocol JSON Exporters'
+ext.moduleName = "io.opentelemetry.exporter.otlpjson"
+
+dependencies {
+    compileOnly project(':sdk:trace')
```

Comment:
@jkwatson So I tried a different approach to splitting all exporters for trace vs metrics here since it seemed like it could be a bit too much. If going with this, I'd revert the change to split out the `otlp` exporters in favor of `compileOnly` dependencies.

One of the reasons it felt like much is we would need to split otproto up as well - they all end up having a common, trace, metrics split. Lots of artifacts.

But it means that previously, a user could only depend on `exporter-otlp` to have the SDK as well, but with such a scheme, they'd have to additionally add the SDK as a dependency. I wonder if this is too annoying or reasonable.

Reply:
So, since the trace SDK is supposed to be stable, I think it would ok to declare this as an `api` dependency and get that for free, like people would probably expect. Keeping the metrics dependency as `provided` might be ok, though, since someone has to explicitly opt-in to depending on it.

Reply:
Do you think so long term? e.g., what if someone replaces the trace SDK with a custom one, would we want the exporter artifacts to still bring in the trace SDK? I guess it could be ok, as custom SDKs still seem rare enough to me. Just want to confirm.

Reply:
If it's a completely custom SDK, it might have a completely different exporter interface, yeah? This does make me think we should have a separate artifact with the "SDK Extension Points" in it, including exporter, span processor, sampler interfaces, so at least custom SDKs can use the same extension points.